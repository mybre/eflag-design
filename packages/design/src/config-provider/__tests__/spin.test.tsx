import React from 'react';
import { render } from '@testing-library/react';
import ConfigProvider from '..';
import Spin from '../../spin';

describe('ConfigProvider spin', () => {
  it('spin.indicator should work', () => {
    const { container, asFragment } = render(
      <ConfigProvider
        spin={{
          indicator: <div className="custom-indicator" />,
        }}
      >
        <Spin />
      </ConfigProvider>
    );
    expect(container.querySelector('.custom-indicator')).toBeTruthy();
    expect(container.querySelector('.ant-spin-eflag')).toBeFalsy();
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('spin.indicator should work in nested ConfigProvider', () => {
    const { container, asFragment } = render(
      <ConfigProvider
        spin={{
          indicator: <div className="custom-indicator-1" />,
        }}
      >
        <ConfigProvider>
          <Spin />
        </ConfigProvider>
        <ConfigProvider
          spin={{
            indicator: <div className="custom-indicator-2" />,
          }}
        >
          <Spin />
        </ConfigProvider>
      </ConfigProvider>
    );
    expect(container.querySelectorAll('.custom-indicator-1').length).toBe(1);
    expect(container.querySelectorAll('.custom-indicator-2').length).toBe(1);
    expect(container.querySelectorAll('.ant-spin-oceanbase').length).toBe(0);
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
