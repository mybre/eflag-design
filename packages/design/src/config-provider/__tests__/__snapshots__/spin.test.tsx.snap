// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigProvider spin spin.indicator should work 1`] = `
<div
  class="ant-app"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-spinning"
  >
    <div
      class="custom-indicator ant-spin-dot"
    />
  </div>
</div>
`;

exports[`ConfigProvider spin spin.indicator should work in nested ConfigProvider 1`] = `
<div
  class="ant-app"
>
  <div
    class="ant-app"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning"
    >
      <div
        class="custom-indicator-1 ant-spin-dot"
      />
    </div>
  </div>
  <div
    class="ant-app"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning"
    >
      <div
        class="custom-indicator-2 ant-spin-dot"
      />
    </div>
  </div>
</div>
`;
